<launch>  
    <!-- Launch Navion  -->
    <include file="$(find nav_launch)/launch/navion_rviz.launch"/>

    <!-- Start braodcaster. This is only necessary since we do not synchronize the transformations yet -->
    <node name="console_broadcaster" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 1 mns console 100" respawn="true" />

    <!-- Arguments -->
    <arg name="host" value="129.132.53.194" />
    <arg name="tcp_role" value="host" />

    <!-- Sockets -->
    <node name="mag_field_socket" pkg="tcp_socket" type="tcp_socket.py" output="screen">
        <param name="host" type="string" value="$(arg host)" />
        <param name="port" type="int" value="3000" />
        <param name="topic_name" type="string" value="/backward_model/field" />
        <param name="message_type" type="string" value="mag_msgs/FieldStamped" />
        <param name="tcp_role" type="string" value="$(arg tcp_role)" />
        <param name="ros_role" type="string" value="publisher" />
    </node>

    <node name="mca_vel_socket" pkg="tcp_socket" type="tcp_socket.py" output="screen">
        <param name="host" type="string" value="$(arg host)" />
        <param name="port" type="int" value="3001" />
        <param name="topic_name" type="string" value="/navion/target_velocity_mca" />
        <param name="message_type" type="string" value="std_msgs/Float32" />
        <param name="tcp_role" type="string" value="$(arg tcp_role)" />
        <param name="ros_role" type="string" value="publisher" />
    </node>

    <node name="actual_field_socket" pkg="tcp_socket" type="tcp_socket.py" output="screen">
        <param name="host" type="string" value="$(arg host)" />
        <param name="port" type="int" value="3002" />
        <param name="topic_name" type="string" value="/navion/actual_field" />
        <param name="message_type" type="string" value="mag_msgs/FieldArrayStamped" />
        <param name="tcp_role" type="string" value="$(arg tcp_role)" />
        <param name="ros_role" type="string" value="subscriber" />
    </node>

    <!-- <node name="basler_camera_socket" pkg="tcp_socket" type="tcp_socket.py" output="screen">
        <param name="host" type="string" value="$(arg host)" />
        <param name="port" type="int" value="3003" />
        <param name="topic_name" type="string" value="/basler_camera/image_raw/compressed" />
        <param name="message_type" type="string" value="sensor_msgs/CompressedImage" />
        <param name="tcp_role" type="string" value="$(arg tcp_role)" />
        <param name="ros_role" type="string" value="subscriber" />
    </node> -->

    <node name="system_report_socket" pkg="tcp_socket" type="tcp_socket.py" output="screen">
        <param name="host" type="string" value="$(arg host)" />
        <param name="port" type="int" value="3004" />
        <param name="topic_name" type="string" value="/navion/system_report" />
        <param name="message_type" type="string" value="nav_controller/systemreport" />
        <param name="tcp_role" type="string" value="$(arg tcp_role)" />
        <param name="ros_role" type="string" value="subscriber" />
    </node>


</launch>